src/main/kotlin/com/matthewrussell/trchunkbrowser/domain/Properties.kt

```diff
@@ -0,0 +1,29 @@
+package com.matthewrussell.trchunkbrowser.domain
+
+import org.apache.commons.configuration2.PropertiesConfiguration
+import org.apache.commons.configuration2.builder.FileBasedConfigurationBuilder
+import org.apache.commons.configuration2.builder.fluent.Configurations
+import java.io.File
+import java.io.FileOutputStream
+
+object Properties {
+    val builder: FileBasedConfigurationBuilder<PropertiesConfiguration>
+    val config: PropertiesConfiguration
+
+    init {
+        val configs = Configurations()
+        val propertiesFile = File("config.properties")
+
+        // Create empty properties file if it doesn't exist
+        if(!propertiesFile.exists()) {
+            val os = FileOutputStream(propertiesFile)
```

Comment:
If this throws an exception, the os.close won't be called. What you'd want to do here is called "try with resources" in Java, where you do something like:
```
try(val os = FileOutputStream(propertiesFile) {
    os.write("".toByteArray())
}
```
This will use the autocloseable interface, so you won't need to call os.close() manually, and in addition, it will automatically deal with putting the close call in a finally block for you internally.

The kotlin extension method that handles this for you is .use { }
so you'd want to do:
```
FileOutputSteam(propertiesFile).use {
    it.write("".toByteArray())
}
```

In fact, you can probably do propertiesFile.outputStream().use { }

Reply:
wow, this so pretty! Updated